File: b/11-07-2022/Mari/Mari/Quadrato.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Mari
{
    internal class Quadrato : Figura
    {
        private double lato;

        public Quadrato() : base(4)
        {
            tipofigura = "Quadrato";
            lato = 0.0;
        }

        public Quadrato(double lato) : base(4)
        {
            if (lato <= 0)
                throw new Exception("Inserito numero non valdio");
            this.lato = lato;
            tipofigura = "Quadrato";
            Area();//RICALCOLO
            Perimetro();
        }

        public double Lato
        {
            get { return lato; }
            set 
            {
                if (value <= 0)
                    throw new Exception("Inserito numero non valdio");
                lato = value;
                Area();//RICALCOLO
                Perimetro();
            }
        }

        private void Area()
        {
            area = lato * lato;
        }

        private void Perimetro()
        {
            perimetro = lato * 4.0;
        }

        public override string Stampa()
        {
            return base.Stampa("\nLato: " +  lato);
        }
    }
}

File: a/12-07-2022/Figure/src/Quadrato.java
public class Quadrato extends Figura{

    private double l;

    public Quadrato()
    {
        nlati = 4;
        l = 0.0;
    }

    public double Area()
    {
        return l * l;
    }

    public double Perimetro()
    {
        return l * 4;
    }

    @Override
    public String ToString()
    {
        return super.ToString() +
                "\nLato: " + l +
                "\nArea: " + Area() +
                "\nPerimetro: " + Perimetro();
    }
}
